src/granite_orm/transactions.cr

```diff
@@ -10,25 +10,31 @@ module Granite::ORM::Transactions
     # The import class method will run a batch INSERT statement for each model in the array
     # the array must contain only one model class
     # invalid model records will be skipped
-    def self.import(model_array : Array(self))
+    def self.import(model_array : Array(self), batch_size : Int32 = model_array.size)
       begin
-        @@adapter.import(table_name, primary_name, fields.dup, model_array)
+        model_array.each_slice(batch_size, true) do |slice|
+          @@adapter.import(table_name, primary_name, fields.dup, slice)
```

Comment:
Can you get away with calling `#dup` on fields only once and reusing that duped copy?

Reply:
I'll take a look, good catch.